### DISCUSSION TITLE
Wrong root folder when building wheel?

### DISCUSSION THREAD

**Discussion by killzoner** [pdm-project/pdm#937] [Q&A] [answered]
- [X] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

## Steps to reproduce

I was testing `pdm build` for a wheel and trying to install it into a conda venv (using `pip install <my_wheel>)`. But everytime I was trying to reference the installed package, it was not working.

I then realized that the root folder for built wheels is probably not right.

If we take https://github.com/pdm-project/pdm-pep517/tree/master/tests/fixtures/projects/demo-module, should not the root folders in the wheel be
- `/demo_module-0.1.0.dist-info/`
- `/demo_module/`

I was testing a few wheels from `pypi` and this structure was always similar.
Instead, it's 
```bash
Reusing shared build env: /tmp/pdm-build-env-50x692ap-shared
 - Adding bar_module.py
 - Adding foo_module.py
 - Adding demo_module-0.1.0.dist-info/WHEEL
 - Adding demo_module-0.1.0.dist-info/METADATA
 - Adding demo_module-0.1.0.dist-info/LICENSE
 - Adding demo_module-0.1.0.dist-info/RECORD
```

## Actual behavior

Package should have a root folder name according to package name

## Expected behavior

Root package is the one from source structure, without root folder being project name.

## Environment Information

```bash
# Paste the output of `pdm info && pdm info --env` below:

PDM version:        1.13.2
Python Interpreter: /home/linuxbrew/.linuxbrew/opt/python@3.9/bin/python3.9 (3.9)
Project Root:       /home/typhoon/pdm-pep517/tests/fixtures/projects/demo-module
Project Packages:   /home/typhoon/pdm-pep517/tests/fixtures/projects/demo-module/__pypackages__/3.9
{
  "implementation_name": "cpython",
  "implementation_version": "3.9.10",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "5.4.0-100-generic",
  "platform_system": "Linux",
  "platform_version": "#113-Ubuntu SMP Thu Feb 3 18:43:29 UTC 2022",
  "python_full_version": "3.9.10",
  "platform_python_implementation": "CPython",
  "python_version": "3.9",
  "sys_platform": "linux"
}
```

Maybe it's just a missing configuration, thanks in advance

**Comment by frostming** [comment, marked as answer]
Taken from another **single module** package:
```
❯ unzip -l pyparsing-2.4.7-py2.py3-none-any.whl
Archive:  pyparsing-2.4.7-py2.py3-none-any.whl
  Length      Date    Time    Name
---------  ---------- -----   ----
   273365  03-31-2020 02:48   pyparsing.py
     1023  04-05-2020 22:19   pyparsing-2.4.7.dist-info/LICENSE
     3636  04-05-2020 22:19   pyparsing-2.4.7.dist-info/METADATA
      110  04-05-2020 22:19   pyparsing-2.4.7.dist-info/WHEEL
       10  04-05-2020 22:19   pyparsing-2.4.7.dist-info/top_level.txt
      468  04-05-2020 22:19   pyparsing-2.4.7.dist-info/RECORD
```
```
❯ unzip -l ../pdm/tests/fixtures/artifacts/future_fstrings-1.2.0-py2.py3-none-any.whl
Archive:  ../pdm/tests/fixtures/artifacts/future_fstrings-1.2.0-py2.py3-none-any.whl
  Length      Date    Time    Name
---------  ---------- -----   ----
      126  06-16-2019 03:04   aaaaa_future_fstrings.pth
     8262  06-16-2019 03:04   future_fstrings.py
     1059  06-16-2019 03:04   future_fstrings-1.2.0.dist-info/LICENSE
     3160  06-16-2019 03:04   future_fstrings-1.2.0.dist-info/METADATA
      110  06-16-2019 03:04   future_fstrings-1.2.0.dist-info/WHEEL
       63  06-16-2019 03:04   future_fstrings-1.2.0.dist-info/entry_points.txt
       16  06-16-2019 03:04   future_fstrings-1.2.0.dist-info/top_level.txt
      686  06-16-2019 03:04   future_fstrings-1.2.0.dist-info/RECORD
---------                     -------
    13482                     8 files
```
This is expected, all files & directories under the wheel root will be extracted and copied to the target path(`site-packages`). This is how a wheel works and PDM isn't breaking the rules. The top-level package name doesn't have to be the same as the PyPI package name, they can be different.

**Comment by killzoner** [comment]
Hello @frostming,

Thanks for the quick answer, I was kind of expecting this, is there any way to achieve the mentioned goal ?
Thing is, installing package from `sdist` works but not from the `wheel`, which is a bit disturbing

**Reply by frostming** [reply]
Can I have the sdist and wheel file?

**Comment by killzoner** [comment]
I ran everything with a fresh local cache of pip on my machine, and I realized both `sdist` and `wheel` are consistent.
So now when installing with the proper folder structure in the sources, I can reference the project. I think i messed up something starting the project and installing a wrong structured package.

Also, one of the missing config here was `includes = ["src/"]` which makes the stuff ok with right structure.

Really sorry to have bothered with this, and I'm really surprised how fast and responsive you are on the issues.
Thanks a bunch.
